app/services/katello/pulp3/repository/docker.rb

```diff
@@ -49,6 +57,36 @@ def distribution_options(path)
             name: "#{backend_object_name}"
           }
         end
+
+        def copy_units_recursively(unit_hrefs, clear_repo = false)
+          tasks = []
+          if clear_repo
+            tasks << create_version(:remove_content_units => ["*"])
+          end
+          tasks << recursive_add_api.create(recursive_manage_class.new(repository: repository_reference.repository_href,
+                                                                       content_units: unit_hrefs))
+          tasks
+        end
+
+        def copy_content_for_source(source_repository, options = {})
+          filters = ContentViewDockerFilter.where(:id => options[:filter_ids])
+          whitelist_ids = []
+          blacklist_ids = []
+          existing_units = source_repository.docker_manifests
+          filters.each do |filter|
+            if filter.inclusion
+              whitelist_ids += filter.get_pulp_ids(source_repository)
+            else
+              blacklist_ids += filter.get_pulp_ids(source_repository)
+            end
+          end
```

Comment:
does this handle the case where i *only* blacklist an item?  Whereby i'd expect everything to be copied except that one item?    It looks like it might copy nothing, but i haven't fully tested it yet

Reply:
It does. I check if there's no whitelist, and if there's none, I copy all tags minus the blacklist files on line 85.

Reply:
oh doh!  i missed that line, looks good!   I will do some testing